src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -134,9 +129,8 @@ protected virtual Task WaitForChangesAsync(RequestContext context, CancellationT
             Contract.ThrowIfNull(context.Solution);
 
             var clientCapabilities = context.GetRequiredClientCapabilities();
-            var category = GetDiagnosticCategory(diagnosticsParams) ?? "";
-            var sourceIdentifier = GetDiagnosticSourceIdentifier(diagnosticsParams) ?? "";
```

Comment:
what happened to sourceIdentifier?  do we not need it?

Reply:
We do not.  It is essentially the same thing as category, just for the public impl of the spec.

Reply:
sourceIdentifier turned into `IDiagnosticSourceProvider.Name`. @dibarbet can explain it better :)